### PR TITLE
Added documentation for PrincipalACl migration and delete-missing-principal cmd

### PR DESCRIPTION
**Pull request by HariGS-DB**
## Changes
Added documentation to explain ACL migration from interactive and SQL warehouse on azure and was for:
 - External Location
 - Catalog and Schema
 - Tables and Views

Added documentation for delete-missing-principal cmd

### Functionality

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [ ] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1236,6 +1237,16 @@ Two optional parameter are available for this command:
 
 [[back to top](#databricks-labs-ucx)]
 
+## `delete-missing-principals` command (AWS Only)
+```bash
+databricks labs ucx delete-missing-principals --aws-profile <aws_profile>
+```
+This command helps to delete the IAM role created by UCX. It lists all the IAM Roles generated by the principal-prefix-access
+cmd and allows user to select multiple roles to delete. It also checks if selected roles are mapped to any storage credentials
```
Comment (JCZuurmond) [inline README.md:1245]:
```suggestion
command and allows user to select multiple roles to delete. It also checks if selected roles are mapped to any storage credentials
```

File: README.md [hunk 2]
```diff
@@ -1311,6 +1322,15 @@ Once the [`assessment` workflow](#assessment-workflow) finished successfully, an
...
 task in the assessment job. You can run [`validate-external-locations` command](#validate-external-locations-command) to check the candidate locations.
 
+Location ACL:
+migrate-locations cmd also applies any location ACL from existing cluster.
+For Azure it checks if there are any interactive cluster or sql warehouse
+which has service principals configured to access storage. It maps the storage account to the external location created and grants CREATE_EXTERNAL_TABLE,
+CREATE_EXTERNAL_VOLUME and READ_FILES permission on the location to all the user who have access to the interactive cluster or sql warehouse
+For AWS, it checks any instance profiles mapped to the interactive cluster or sql warehouse. It checks the mapping of instance profiles to the bucket. It then
+maps the bucket to the external locations created and grants CREATE_EXTERNAL_TABLE, CREATE_EXTERNAL_VOLUME and READ_FILES permission on the location to all the user who have access to the interactive cluster
+or sql warehouse
```
Comment (JCZuurmond) [inline README.md:1332]:
```suggestion
**Location ACLs:**
`migrate-locations` command applies any location ACL from existing cluster.
For Azure, it checks if there are any interactive cluster or SQL warehouse
which has service principals configured to access storage. It maps the storage account to the external location created and grants `CREATE_EXTERNAL_TABLE`,
`CREATE_EXTERNAL_VOLUME` and `READ_FILES` permission on the location to all the user who have access to the interactive cluster or SQL warehouse
For AWS, it checks any instance profiles mapped to the interactive cluster or SQL warehouse. It checks the mapping of instance profiles to the bucket. It then
maps the bucket to the external locations created and grants `CREATE_EXTERNAL_TABLE`, `CREATE_EXTERNAL_VOLUME` and `READ_FILES` permission on the location to all the user who have access to the interactive cluster
or SQL warehouse
```

File: README.md [hunk 3]
```diff
@@ -1374,7 +1394,14 @@ databricks labs ucx create-catalogs-schemas
 ```
 After [`create-table-mapping` command](#create-table-mapping-command) is executed, you can run this command to have the required UC catalogs and schemas created.
 This command is supposed to be run before migrating tables to UC using [table migration process](#Table-Migration).
-
+Catalog & Schema ACL:
```
Comment (JCZuurmond) [inline README.md:1397]: Similar changes as above

**Review by JCZuurmond — COMMENTED** [review]
some nits

File: README.md (hunk 3 above)
Comment (HariGS-DB) [reply README.md:1399]: Done

**Pull request merged**